lib/govuk_index/publishing_event_message_handler.rb

```diff
@@ -15,57 +13,56 @@ class NotIdentifiable < StandardError; end
 
   class MissingExternalUrl < StandardError; end
 
-  DOCUMENT_TYPES_WITHOUT_BASE_PATH =
-    %w[
-      contact
-      role_appointment
-      world_location
-
-      #
-      role
-      document
-      types
-      ambassador_role
-      board_member_role
-      chief_professional_officer_role
-      chief_scientific_officer_role
-      chief_scientific_advisor_role
-      deputy_head_of_mission_role
-      governor_role
-      high_commissioner_role
-      military_role
-      ministerial_role
-      special_representative_role
-      traffic_commissioner_role
-      worldwide_office_staff_role
-    ].freeze
-
-  class PublishingEventJob < BaseJob
-    notify_of_failures
-
-    def perform(messages)
-      processor = Index::ElasticsearchProcessor.govuk
+  class PublishingEventMessageHandler
+    DOCUMENT_TYPES_WITHOUT_BASE_PATH =
+      %w[
+        contact
+        role_appointment
+        world_location
+
+        #
+        role
+        document
+        types
+        ambassador_role
+        board_member_role
+        chief_professional_officer_role
+        chief_scientific_officer_role
+        chief_scientific_advisor_role
+        deputy_head_of_mission_role
+        governor_role
+        high_commissioner_role
+        military_role
+        ministerial_role
+        special_representative_role
+        traffic_commissioner_role
+        worldwide_office_staff_role
+      ].freeze
+
+    def initialize(routing_key, payload)
+      @routing_key = routing_key
+      @payload = payload
+      @logger = Logging.logger[self]
+    end
 
-      messages.each do |routing_key, payload|
-        process_action(processor, routing_key, payload)
-      end
+    def self.process(...)
+      new(...).process
+    end
 
-      responses = processor.commit
+    def process
```

Comment:
Sorry I think a comment I made last time got lost (or I messed it up somehow).

I think we should rename these methods to be `call` rather than process to have a more generic entrypoint.